ksqldb-functional-tests/src/test/resources/query-validation-tests/joins.json

```diff
@@ -3754,6 +3754,246 @@
+        "CREATE STREAM SOURCE WITH (kafka_topic='source', format='AVRO', KEY_SCHEMA_ID=1, KEY_SCHEMA_FULL_NAME='com.test.namespace.keySchema', VALUE_SCHEMA_ID=2, VALUE_SCHEMA_FULL_NAME='com.test.namespace.valueSchema');",
+        "CREATE TABLE SINK WITH (kafka_topic='source2', format='AVRO');",
+        "CREATE STREAM OUT AS SELECT S.*, SS.* FROM SOURCE S JOIN SINK SS ON STRUCT(K := S.B)=SS.ROWKEY;"
+      ],
+      "topics": [
+        {
+          "name": "source",
+          "keyFormat": "AVRO",
+          "keySchema": {"type":"record","namespace":"com.test.namespace","name":"keySchema","fields":[{"name":"K","type":"string"}]},
+          "valueFormat": "AVRO",
+          "valueSchema": {"type":"record","namespace":"com.test.namespace","name":"valueSchema","fields":[{"name":"A","type":"int"},{"name":"B","type":"string"}]}
+        },
+        {
+          "name": "source2",
+          "keyFormat": "AVRO",
+          "keySchema": {"type":"record","namespace":"com.test.namespace","name":"keySchema","fields":[{"name":"K","type":"string"}]},
+          "valueFormat": "AVRO",
+          "valueSchema": {"type":"record","namespace":"com.test.namespace","name":"valueSchema","fields":[{"name":"A","type":"int"},{"name":"B","type":"string"}]}
+        }
+      ],
+      "inputs": [
+        {"topic": "source2", "key": {"K": "falcon"}, "value": {"A": 123, "B": "falcon"}, "timestamp": 0},
+        {"topic": "source2", "key": {"K": "giraffe"}, "value": {"A": 456, "B": "giraffe"}, "timestamp": 0},
+        {"topic": "source2", "key": {"K": "turtle"}, "value": {"A": 789, "B": "turtle"}, "timestamp": 0},
+        {"topic": "source", "key": {"K": "0"}, "value": {"A": 123, "B": "falcon"}, "timestamp": 0},
+        {"topic": "source", "key": {"K": "0"}, "value": {"A": 456, "B": "giraffe"}, "timestamp": 0},
+        {"topic": "source", "key": {"K": "0"}, "value": {"A": 789, "B": "turtle"}, "timestamp": 0},
+        {"topic": "source2", "key": {"K": "turtle"}, "value": {"A": 800, "B": "turtle"}, "timestamp": 0}
+      ],
+      "outputs": [
+        {"topic": "OUT", "key": {"K": "falcon"}, "value": {"S_ROWKEY": {"K": "0"}, "S_A": 123, "S_B": "falcon", "SS_A": 123, "SS_B": "falcon"}, "timestamp": 0},
+        {"topic": "OUT", "key": {"K": "giraffe"}, "value": {"S_ROWKEY": {"K": "0"}, "S_A": 456, "S_B": "giraffe", "SS_A": 456, "SS_B": "giraffe"}, "timestamp": 0},
+        {"topic": "OUT", "key": {"K": "turtle"}, "value": {"S_ROWKEY": {"K": "0"}, "S_A": 789, "S_B": "turtle", "SS_A": 789, "SS_B": "turtle"}, "timestamp": 0}
+      ],
+      "post": {
+        "sources": [
+          {
+            "name": "OUT",
+            "type": "stream",
+            "keyFormat": {"format": "AVRO", "properties": {"fullSchemaName": "io.confluent.ksql.avro_schemas.OutKey"}, "features": ["UNWRAP_SINGLES"]},
+            "schema": "SS_ROWKEY STRUCT<K STRING> KEY, S_ROWKEY STRUCT<K STRING>, S_A INTEGER , S_B STRING, SS_A INTEGER, SS_B STRING"
+          }
+        ]
+      }
```

Comment:
Thanks for the test coverage! Can we add a quick test for stream-stream joins, too, for completeness? Also, you can add a test with JSON_SR (the third SR-enabled key format supported today) in the process. (No need to go back and add JSON_SR coverage for all the other test cases you've already added, though, since I don't think it should be fundamentally different from Avro.)